src/mixins/itext_key_behavior.mixin.js

```diff
@@ -579,7 +579,6 @@ fabric.util.object.extend(fabric.IText.prototype, /** @lends fabric.IText.protot
 
     this._removeExtraneousStyles();
 
-    this._clearCache();
```

Comment:
this gets removed.
Removing char is changing text property so no need to enforce cache clearing.